src/lib/sensor_calibration/Magnetometer.hpp

```diff
@@ -85,6 +85,12 @@ class Magnetometer
 		return _rotation * (_scale * ((data + _power * _power_compensation) - _offset));
 	}
 
+	// Compute sensor offset from bias (board frame)
+	matrix::Vector3f BiasCorrectedSensorOffset(const matrix::Vector3f &bias) const
+	{
+		return _scale.I() * _rotation.I() * bias + _offset;
```

Comment:
Is this equation correct? I think it is, but can't find documentation for which frame of reference the class variables _scale, _rotation and _offset are defined in and the order that rotations and offsets should be applied. This function implies that bias and _offset are in board frame, _rotation rotates from sensor frame to board frame and _scale removes scale and soft iron errors in sensor frame after removal of offsets.

Reply:
This `Magnetometer` class (src/lib/sensor_calibration/Magnetometer.hpp) is where I've collected all the corrections applied to raw sensor data that's been ripped out of the drivers and sensors module. All sensors now publish raw data (sensor_accel/gyro/mag) and everything is applied downstream. The only exception is for internal sensors that are started by the board with a particular rotation. 

Ultimately this is what's applied to each raw sample. https://github.com/PX4/PX4-Autopilot/blob/64240010bb67b75f880a4972189dadd988055b17/src/lib/sensor_calibration/Magnetometer.hpp#L83-L86

So yes, we should really do a pass to thoroughly document everything.